README.md

```diff
@@ -106,6 +125,13 @@ Current limitations for using custom sharding key:
 - `primary_index_fieldno_map` is not cached, see
   [#243](https://github.com/tarantool/crud/issues/243).
 
+Current limitations for using custom sharding functions:
+
+- It's not possible to update sharding functions automatically when schema is
+  updated on storages, see
+  [#212](https://github.com/tarantool/crud/issues/212). However it is possible
+  to do it manually with `require('crud.common.sharding_func').update_cache()`.
```

Comment:
Sadly, we didin't think on it before (my bad).

We should leave `require('crud.common.sharding_key').update_cache()` working and we should not add more hacky intructions to our customers to make things 'just work'. IOW, `require('crud.common.sharding_key').update_cache()` should renew all caches.

I hope we'll fix it soon and it'll be no-op.

Reply:
`require('crud.common.sharding_key').update_cache()` updates both caches, you can see it in `_fetch_on_router` function:

```lua
local _fetch_on_router = locked(function(timeout, space_name, metadata_map_name)
    dev_checks('number', 'string', 'string')

    if cache[metadata_map_name] ~= nil then
        return
    end

    local metadata_map, err = call.any(FETCH_FUNC_NAME, {}, {
        timeout = timeout
    })
    if err ~= nil then
        return err
    end
    if metadata_map == nil then
        cache[cache.SHARDING_KEY_MAP_NAME] = {}
        cache[cache.SHARDING_FUNC_MAP_NAME] = {}
        return
    end

    local err = sharding_key.construct_as_index_obj_cache(metadata_map, space_name)
    if err ~= nil then
        return err
    end

    local err = sharding_func.construct_as_callable_obj_cache(metadata_map, space_name)
    if err ~= nil then
        return err
    end
end)
```

But it returns only sharding key for specified space, I highlighted it in README:

```diff
  to do it manually with `require('crud.common.sharding_key').update_cache()`
+ (this function updates both caches: sharding key cache and sharding function
+ cache, but returned value is sharding key from cache).
```

Similarly for sharding func:

```diff
  to do it manually with `require('crud.common.sharding_func').update_cache()`
+ (this function updates both caches: sharding key cache and sharding function
+ cache, but returned value is sharding function from cache).
```